charts/openstack-cloud-controller-manager/values.yaml

```diff
@@ -63,3 +65,24 @@ cloudConfig:
   loadBalancer:
   blockStorage:
   metadata:
+
+# Additional volumes that should be mounted into the pods
+# The following three volumes are required to use all OCCM controllers,
+# but might not be needed if you just use a specific controller
+extraVolumeMounts:
```

Comment:
What about leaving the extra mounts empty by default?

Like this, `/usr/libexec/kubernetes/kubelet-plugins/volume/exec` can not be removed.


Have you considered the approach below? To leave it empty and let users pick what they need?
https://github.com/kubernetes/ingress-nginx/blob/ab4fa4246797a33bc7b45b45097a91086fe9ad48/charts/ingress-nginx/values.yaml#L537-L545

Reply:
Leaving it empty would be a backwards-incompatible change.
You can override the entire array by specifying `extraVolumeMounts: []` (or adding the desired entries only).

Reply:
It seems that it doesn't merge the default extraVolumeMounts and the user-defined ones. We can override the defaults, thanks!

Is this syntax used elsewhere? I still think it would be friendlier to use standard kubernetes syntax for volumes and volumeMounts.
```
  - name: flexvolume-dir
    hostPath: /usr/libexec/kubernetes/kubelet-plugins/volume/exec
    mountPath: /usr/libexec/kubernetes/kubelet-plugins/volume/exec
```

Reply:
What do you mean by "standard kubernetes syntax"?

Reply:
In the examples below volumes and volumeMounts are defined as they would in kubernetes manifest. Not sure how else to explain "standard kubernetes syntax".

* https://github.com/kubernetes/cloud-provider-openstack/blob/a9da92cddac704db6d940abbc8f1b2dc20b673cc/charts/cinder-csi-plugin/values.yaml#L52-L62
* mentioned above https://github.com/kubernetes/ingress-nginx/blob/ab4fa4246797a33bc7b45b45097a91086fe9ad48/charts/ingress-nginx/values.yaml#L537-L545

Reply:
I see, thanks for the example. Well, I tried to adapt the data structure to this use-case because I don't like exposing all the implementation details (otherwise the user eventually needs to pass in the entire `spec` structure).

Reply:
Hi @jacksgt, thanks for your contribution! This really seems a bit over-complicated IMO. We'd need to keep around the comment documenting various volume types and options. We'd also lose input validation (imagine making a typo `subpath` instead of `subPath`).

I think it's best to stick with what's already in Kubernetes if we can. kubernetes/ingress-nginx seems like a good example to follow, having `extraVolumes` and `extraVolumeMounts` fields. Not only is this approach more flexible for users, we keep all the input validation. Maybe adding a comment referring users to the official Kubernetes docs on what's the syntax for volumes and volume mounts would help too.

Reply:
Alrighty, thanks for the feedback. I have updated the PR accordingly.
Would you mind taking a look? @gman0 @strigazi

Reply:
@jacksgt great, thank you!